src/Microsoft.ML.Data/Evaluators/EvaluatorStaticExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.ML.Data.StaticPipe;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime.Training;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// Extension methods for evaluation.
+    /// </summary>
+    public static class EvaluatorStaticExtensions
+    {
+        /// <summary>
+        /// Evaluates scored binary classification data.
+        /// </summary>
+        /// <typeparam name="T">The shape type for the input data.</typeparam>
+        /// <param name="ctx">The binary classification context.</param>
+        /// <param name="data">The data to evaluate.</param>
+        /// <param name="label">The index delegate for the label column.</param>
+        /// <param name="pred">The index delegate for columns from calibrated prediction of a binary classifier.
+        /// Under typical scenarios, this will just be the same tuple of results returned from the trainer.</param>
+        /// <returns>The evaluation results for these calibrated outputs.</returns>
+        public static BinaryClassifierEvaluator.CalibratedResult Evaluate<T>(
+            this BinaryClassificationContext ctx,
```

Comment:
Why are some Evaluate methods on `BinaryClassificationContext` directly. And these are extension methods?

Reply:
There was a desire I think to make PiGSTy "separable." While I am not separating it yet, I am trying to write my code in such a way that it *can* be separated, and I could not do that if I didn't make these pigsty things extension methods. On the other hand there was no reason to make the dynamic API based methods extension methods, so I did not.